### PR TITLE
Pr/fix compile warnings

### PR DESCRIPTION
**Pull request by uli42**
fixes some warnings, please review

apply on top of #222

### CODE REVIEW

File: nx-X11/programs/Xserver/Xext/xcmisc.c [hunk 1]
```diff
@@ -40,11 +40,7 @@ from The Open Group.
 #include <nx-X11/extensions/xcmiscstr.h>
 #include "modinit.h"
 
-#if HAVE_STDINT_H
 #include <stdint.h>
-#elif !defined(UINT32_MAX)
-#define UINT32_MAX 0xffffffffU
-#endif
```
Comment (sunweaver) [inline nx-X11/programs/Xserver/Xext/xcmisc.c:47]: There is a typo in the commit msg of 6b9d3db43626249672738dbd54ce5d7a9688f315: definde -> defined

**Review by sunweaver — APPROVED** [review]
I left one comment on this PR (typo in one commit msg). The rest looks good.

Setting this PR to approved, but please fix that commit msg when finally rebasing.

**Comment by sunweaver** [comment]
@uli42: Please rebase this PR branch, it is the next in my merging queue...

**Comment by uli42** [comment]
I have added some more fixes and rebased this.

**Pull request merged**